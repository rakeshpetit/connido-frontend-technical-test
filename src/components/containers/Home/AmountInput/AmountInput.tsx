import React, { Component } from 'react';
import { View, TextInput, StyleSheet } from 'react-native';

type Props = {
  calculatedSourceAmount: number,
  calculatedDestinationAmount: number,
  setUserSourceAmountDone: Function,
  setUserDestinationAmountDone: Function
};

class AmountInput extends Component<Props> {
  onSourceAmountChange = (text: any) => {
    if(!isNaN(text)) {
      this.props.setUserSourceAmountDone(text);
    }
  }
  onDestinationAmountChange = (text: any) => {
    if(!isNaN(text)) {
      this.props.setUserDestinationAmountDone(text);
    }
  }
  render() {
    const { calculatedSourceAmount, calculatedDestinationAmount } = this.props;
    return (
      <View style={styles.container}>
        <TextInput onChangeText={(text: string) => this.onSourceAmountChange(text)}
          keyboardType={'numeric'} style={styles.textInput}>
          {calculatedSourceAmount ? calculatedSourceAmount : ''}</TextInput>
        <TextInput onChangeText={(text: string) => this.onDestinationAmountChange(text)} 
        keyboardType={'numeric'} style={styles.textInput}>
          {calculatedDestinationAmount ? calculatedDestinationAmount : ''}</TextInput>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    backgroundColor: '#ECF9E0',
    width: '100%',
    padding: 10
  },
  textInput: {
    alignItems: 'center',
    marginTop: 20,
    marginBottom: 10,
    borderBottomColor: '#000000',
    backgroundColor: 'white',
    borderWidth: 1,
    padding: 10,
    fontSize: 20,
    width: '45%'
  }
});

export default AmountInput;
