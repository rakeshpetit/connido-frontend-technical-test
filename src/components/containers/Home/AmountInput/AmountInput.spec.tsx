import React from 'react';
import renderer from 'react-test-renderer';

import AmountInput from './AmountInput';

it('renders correctly when there are no items', () => {
  const calculatedSourceAmount = 100;
  const calculatedDestinationAmount = 89.95;
  const setUserSourceAmountDone = () => {};
  const setUserDestinationAmountDone = () => {};
  const tree = renderer.create
    (<AmountInput 
        calculatedSourceAmount={calculatedSourceAmount}
        calculatedDestinationAmount={calculatedDestinationAmount}
        setUserSourceAmountDone={setUserSourceAmountDone}
        setUserDestinationAmountDone={setUserDestinationAmountDone}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});