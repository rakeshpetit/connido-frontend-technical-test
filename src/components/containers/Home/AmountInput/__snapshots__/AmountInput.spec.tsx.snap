// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AmountInput correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ECF9E0",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#000000",
        "borderWidth": 1,
        "fontSize": 20,
        "marginBottom": 10,
        "marginTop": 20,
        "padding": 10,
        "width": "45%",
      }
    }
  >
    100
  </TextInput>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#000000",
        "borderWidth": 1,
        "fontSize": 20,
        "marginBottom": 10,
        "marginTop": 20,
        "padding": 10,
        "width": "45%",
      }
    }
  >
    89.95
  </TextInput>
</View>
`;

exports[`renders correctly when there are no items 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ECF9E0",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#000000",
        "borderWidth": 1,
        "fontSize": 20,
        "marginBottom": 10,
        "marginTop": 20,
        "padding": 10,
        "width": "45%",
      }
    }
  >
    100
  </TextInput>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#000000",
        "borderWidth": 1,
        "fontSize": 20,
        "marginBottom": 10,
        "marginTop": 20,
        "padding": 10,
        "width": "45%",
      }
    }
  >
    89.95
  </TextInput>
</View>
`;
